projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -514,9 +520,16 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
             return;
         }
         this.selectionService.clear(true);
-        this.onPagingDone.emit({ previous: this._page, current: val });
+
+        const eventArgs = { previous: this._page, current: val };
+        const cancelableEventArgs = { ...eventArgs, cancel: false, owner: this };
+        this.onPaging.emit(cancelableEventArgs);
+
+        if (cancelableEventArgs.cancel) { return; }
+
         this._page = val;
         this.pageChange.emit(this._page);
```

Comment:
Shall we annotate that event as deprecated?

Reply:
Actually, I have wrongly indicated `pageChange` to be deprecated. It is an event to serve the two way data binding of the `page` property. I will correct the table with changes in the issue description now. Good catch.

Reply:
shaw we document it as hidden so it is not presented into the API doc?

Reply:
Well now I see it is indeed public. I agree we should hide it, so I will address this into the changes.